compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -633,28 +637,43 @@ class ComposeScene internal constructor(
             // If pressOwner is under focusedOwner, hover state must be updated
             owner = null
         }
+        if (processHover(event, owner)) {
+            return
```

Comment:
It is subjective, but let's avoid return-in-the-middle, if we can in the future. It can complicate code pathes comparing to small if's.

Reply:
Disagree here. It reduces indention and nesting so makes the code more readable.

What's true here is that this kind of "pre" checks works better near the function start, but anyway

Reply:
> indention and nesting

extracting to a separate function works better in this case

> What's true here is that this kind of "pre" checks works better near the function start

checks at the start are okay.

Reply:
In this particular case this function handles hover and move. So the check is like

```kt
if (consumed) {
    return
}
// Do next action
```

As an improvement I can suggest adding explicit `consumed` variable, but not inverting the condition